api/database/factories/WorkExperienceFactory.php

```diff
@@ -26,15 +38,83 @@ class WorkExperienceFactory extends Factory
     public function definition()
     {
         $startDate = $this->faker->date();
+        $employmentCategory = $this->faker->randomElement(EmploymentCategory::cases())->name;
```

Comment:
It might be possible to tidy some of this stuff out of the front of the `definition` function by using closures.  For example, if you check api/database/factories/DigitalContractingQuestionnaireFactory.php you can see how `authorities_involved` gets set to an enum and then `authority_involved_other` gets conditionally set in a closure based on that value without needing an _if_ structure before the attributes array.

Reply:
Oh, that's a pretty neat method to do this. Thanks. I think I got it all going in https://github.com/GCTC-NTGC/gc-digital-talent/pull/11947/commits/012e8e2d580e4ad6ab3d0ae290726b65b0268dc9

Reply:
Oh, so much nicer!  :chefs-kiss: